graphql_compiler/query_pagination/pagination_planning.py

```diff
@@ -26,14 +55,17 @@ class PaginationPlan(NamedTuple):
     vertex_partitions: Tuple[VertexPartition, ...]
 
 
-# A field with N quantiles has a pagination capacity N / <QUANTILE_REQUIREMENT_FACTOR>
-QUANTILE_REQUIREMENT_FACTOR = 5
+def get_pagination_plan(
+    schema_info: QueryPlanningSchemaInfo, query_ast: DocumentNode, number_of_pages: int
+):
+    """Make a PaginationPlan for the given query and number of desired pages if possible.
 
+    Raises GraphQLPaginationError if the statistics object is misconfigured.
 
-# TODO(bojanserafimov): Make this function return a best effort pagination plan
-#                       when a good one is not found instead of returning None.
-def get_pagination_plan(schema_info, query_ast, number_of_pages):
-    """Make a PaginationPlan for the given query and number of desired pages if possible."""
+    Might paginate to fewer than the desired number of pages if no good pagination plan
```

Comment:
`Might` or `will`? Under what circumstance other than the lack of a good pagination plan will this happen?

Reply:
Might. I elaborated more on when this happens.